### PR TITLE
Adding test/impl tags for reactors

### PR DESCRIPTION
**Pull request by bdlafleur**
This PR adds test/impl tags and test for reactor.py requirements.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by bdlafleur** [comment]
There is no new source code added that would reduce coverage. Attributing failed checks to source spray.

**Comment by john-science** [comment]
Ignore the code coverage diff.

The service we use (coveralls.io) is an idiot. Someone has a PR out that increases code coverage from `main`, and coveralls can't tell the difference between `main` and a PR branch.

File: armi/reactor/tests/test_reactors.py [hunk 1]
```diff
@@ -237,6 +237,13 @@ def setUp(self):
         )
 
     def test_factorySortSetting(self):
+        """
+        Create a core object from an input yaml.
+
+        .. test:: Create core object from input yaml
+            :id: T_ARMI_R_CORE
```
Comment (john-science) [inline armi/reactor/tests/test_reactors.py:244]: I love it.

But looking at the req, I think we need this line somewhere in the test:

```python
self.assertTrue(isinstance(r0.core, Composite))
```

which means we need something like this import at the time:

```python
from armi.reactor.composites import Composite
```

**Review by john-science — APPROVED** [review]
Great!

**Pull request merged**